example/example.cpp

```diff
@@ -24,507 +24,627 @@ using namespace isaac;
+    }
+
+    ISAAC_HOST_DEVICE_INLINE ElemType getPosition() const
+    {
+        return *current_element;
+    }
+
+    ISAAC_HOST_DEVICE_INLINE isaac_float_dim<feature_dim> getAttribute() const
+    {
+        return {isaac_float ( local_grid_coord.x ), isaac_float ( local_grid_coord.y ), isaac_float ( local_grid_coord.z ) };
+    }
+
+    ISAAC_HOST_DEVICE_INLINE isaac_float getRadius() const
+    {
+        return 0.05f;
+    }
+
+
+private:
+    ElemType* current_element;
+    isaac_uint3 local_grid_coord;
+
+};
+
+/*
+ * Particle source
+ */
+ISAAC_NO_HOST_DEVICE_WARNING
+#if ISAAC_ALPAKA == 1
+template < typename TDevAcc, typename THost, typename TStream >
+#endif
+class ParticleSource1
+{
+public:
+    static const ISAAC_IDX_TYPE feature_dim = 3;
+    ISAAC_NO_HOST_DEVICE_WARNING
+    ParticleSource1 (
+#if ISAAC_ALPAKA == 1
+        TDevAcc acc,
+        THost host,
+        TStream stream,
+#endif
+        isaac_float3* ptr,
+        size_t size
+    ) :
+        ptr ( ptr ),
+        size ( size )
+    {}
+
+    ISAAC_HOST_INLINE static std::string getName()
+    {
+        return std::string ( "Particle Source 1" );
+    }
+
+    ISAAC_HOST_INLINE void update ( bool enabled, void* pointer ) {}
+
+    isaac_float3* ptr;
+    size_t size;
+
+
+    // Returns correct particle iterator for the requested cell (in the example the same particle list for each cell)
+    ISAAC_NO_HOST_DEVICE_WARNING
+    ISAAC_HOST_DEVICE_INLINE ParticleIterator1<feature_dim, isaac_float3> getIterator ( const isaac_uint3& local_grid_coord ) const
+    {
+
+        return ParticleIterator1<feature_dim, isaac_float3> ( ptr, size, local_grid_coord );
+    }
 };
 
 
 //////////////////
 // Main program //
 //////////////////
-int main(int argc, char **argv)
+int main ( int argc, char **argv )
 {
-	//Settings the parameters for the example
-	char __server[] = "localhost";
-	char* server = __server;
-	char* filename = NULL;
-	//If existend first parameter is the server. Default: "localhost"
-	if (argc > 1)
-		server = argv[1];
-	int port = 2460;
-	//If existend second parameter is the port. Default: 2460
-	if (argc > 2)
-		port = atoi(argv[2]);
-	if (argc > 3)
-		filename = argv[3];
-
-	//MPI Init
-	int rank,numProc;
-	MPI_Init(&argc, &argv);
-	MPI_Comm_rank(MPI_COMM_WORLD, &rank);
-	MPI_Comm_size(MPI_COMM_WORLD, &numProc);
-
-	//Let's calculate the best spatial distribution of the dimensions so that d[0]*d[1]*d[2] = numProc
-	ISAAC_IDX_TYPE d[3] = {1,1,1};
-	recursive_kgv(d,numProc,2);
-	ISAAC_IDX_TYPE p[3] = { rank % d[0], (rank / d[0]) % d[1],  (rank / d[0] / d[1]) % d[2] };
-
-	//Let's generate some unique name for the simulation and broadcast it
-	int id;
-	if (rank == 0)
-	{
-		srand(time(NULL));
-		id = rand() % 1000000;
-	}
+    //Settings the parameters for the example
+    char __server[] = "localhost";
+    char* server = __server;
+    char* filename = NULL;
+    //If existend first parameter is the server. Default: "localhost"
+    if ( argc > 1 ) {
```

Comment:
could you remove `{}` https://github.com/ComputationalRadiationPhysics/contributing/blob/master/codingGuideLines/cpp.md#12-if-statements